test/files/neg/warn-unused-params.scala

```diff
@@ -94,7 +94,7 @@ trait Anonymous {
 
   def f2: Int => Int = _ + 1  // no warn placeholder syntax (a fresh name and synthetic parameter)
 
-  def g = for (i <- List(1)) yield answer    // warn map.(i => 42)
+  def g = for (i <- List(1)) yield answer    // no warn patvar elaborated as map.(i => 42)
```

Comment:
can you elaborate on this change..?

Reply:
per the end of my comment above, this used to warn under "params" but now warns under "patvars" (because `i` is a pattern var).